sdl_android/src/main/java/com/smartdevicelink/transport/MultiplexTcpTransport.java

```diff
@@ -0,0 +1,410 @@
+/*
+ * Copyright (c) 2018 Livio, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Livio Inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.smartdevicelink.transport;
+
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Message;
+import android.util.Log;
+
+import com.smartdevicelink.protocol.SdlPacket;
+import com.smartdevicelink.transport.enums.TransportType;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+
+import static com.smartdevicelink.util.DebugTool.logError;
+import static com.smartdevicelink.util.NativeLogTool.logInfo;
+
+public class MultiplexTcpTransport extends MultiplexBaseTransport {
+
+	private static final String TAG = "MultiplexTcpTransport";
+
+	private static final int READ_BUFFER_SIZE = 4096;
+	private static final int RECONNECT_DELAY = 5000;
+	private static final int RECONNECT_RETRY_COUNT = 30;
+
+    final String ipAddress;
+    final int port;
+    final boolean autoReconnect;
+
+	private Socket mSocket = null;
+	private InputStream mInputStream = null;
+	private OutputStream mOutputStream = null;
+	private MultiplexTcpTransport.TcpTransportThread mThread = null;
+	private WriterThread writerThread;
+
+
+	public MultiplexTcpTransport(int port, String ipAddress, boolean autoReconnect, Handler handler) {
```

Comment:
since this is never instantiated directly by the developer (non-primary transport), does it need to be public?

Reply:
This will likely need to remain public as it will need to be instantiated as a debugging/development transport and the other TCP transport be removed.